### ISSUE TITLE
Feature request: locks per clone

### ISSUE THREAD

**Issue by xmedeko** [git-lfs/git-lfs#5562]
**Describe the issue**

I've cloned the same repository on 2 machines with the **same username**. I've locked a file on machine A, then I've modified the same file on machine B, commit it and pushed it. So, when I go back to machine A, I can have a conflict during `git pull`, although I've locked the file there.

Also, `git lfs locks --verify` output is same on both machines.

I would like to have the locks to be bound to a clone (working tree) the **same way like SVN** binds locks to it's working copy.

A solution may be to generate UUID (GID) on `git lfs init` (or the first lock), store it in `.git/lfs/...` and use this UUID to distinguish if the lock belongs to the clone or not.

**Comment by chrisd8088** [comment]
The Git LFS [locking API](https://github.com/git-lfs/git-lfs/blob/59194adc4582c30f7012a4f3d1efce859e9bdb4b/docs/api/locking.md) explains the current set of required and optional fields in communications with the server.  There's also this note regarding the [creation](https://github.com/git-lfs/git-lfs/blob/59194adc4582c30f7012a4f3d1efce859e9bdb4b/docs/api/locking.md#create-lock) of locks:

> Servers should ensure that users have push access to the repository, and that files are locked exclusively to one user.

I suppose my question regarding the idea of "local" locks would be, would these prevent other users from operating on the files?  If so, then they can't really be purely local and have to be communicated to the server.  In this case, the user's ID alone is insufficient because you're trying to distinguish between your own locks in different working trees (i.e., one can lock a file in clone A, and then be prevented from touching that file in clone B, even while using the same user account).  I don't quite see how to make that work with a purely local client-side change, but if you want to write up a proposal and show how it could function, we're always interested in new ideas!

**Comment by xmedeko** [comment]
Ad **Solution 2**: yes, the server would keep the lock info, but **clients only** would decide if a lock belongs to the local working tree (i.e. it's possible to edit the locked file) or belong to other working tree (and maybe user, too, i.e. it's not possible to edit the file). 

Looking at the API [List Locks for Verificatio](https://github.com/git-lfs/git-lfs/blob/main/docs/api/locking.md#list-locks-for-verification) seem to me Git LFS locking relies on the server to decide `ours` and `theirs` locks. So, this change would not fit to the current Git LFS locking architecture and Solution 1 is more appropriate?

**Comment by chrisd8088** [comment]
> Git LFS locking relies on the server to decide `ours` and `theirs` locks.

Yes, that's the case with the current API.  So a proposal could either build on that, as with your first proposed idea, or try to expand it in some way to allow for more client-side determinations.  I don't have a strong preference at the moment; I think it would be valuable to see a worked-out proposal and then thenvarious implementors of Git LFS services could review and discuss it.
